platform/ios/app/MBXViewController.m

```diff
@@ -512,39 +463,42 @@ - (void)performActionForSettingAtIndexPath:(NSIndexPath *)indexPath
                     [self.mapView resetPosition];
                     break;
                 case MBXSettingsDebugToolsTileBoundaries:
-                    self.mapView.debugMask ^= MGLMapDebugTileBoundariesMask;
+                    self.currentState.debugMask ^= MGLMapDebugTileBoundariesMask;
                     break;
                 case MBXSettingsDebugToolsTileInfo:
-                    self.mapView.debugMask ^= MGLMapDebugTileInfoMask;
+                    self.currentState.debugMask ^= MGLMapDebugTileInfoMask;
                     break;
                 case MBXSettingsDebugToolsTimestamps:
-                    self.mapView.debugMask ^= MGLMapDebugTimestampsMask;
+                    self.currentState.debugMask ^= MGLMapDebugTimestampsMask;
                     break;
                 case MBXSettingsDebugToolsCollisionBoxes:
-                    self.mapView.debugMask ^= MGLMapDebugCollisionBoxesMask;
+                    self.currentState.debugMask ^= MGLMapDebugCollisionBoxesMask;
                     break;
                 case MBXSettingsDebugToolsOverdrawVisualization:
-                    self.mapView.debugMask ^= MGLMapDebugOverdrawVisualizationMask;
+                    self.currentState.debugMask ^= MGLMapDebugOverdrawVisualizationMask;
                     break;
                 case MBXSettingsDebugToolsShowZoomLevel:
                 {
-                    self.mapInfoHUDEnabled = !self.mapInfoHUDEnabled;
-                    self.hudLabel.hidden = !self.mapInfoHUDEnabled;
+                    self.zoomLevelOrnamentEnabled = !self.zoomLevelOrnamentEnabled;
+                    self.currentState.showsZoomLevelOrnament = self.zoomLevelOrnamentEnabled;
+                    self.hudLabel.hidden = !self.zoomLevelOrnamentEnabled;
+
                     self.reuseQueueStatsEnabled = NO;
                     [self updateHUD];
                     break;
                 }
                 case MBXSettingsDebugToolsShowFrameTimeGraph:
                 {
                     self.frameTimeGraphEnabled = !self.frameTimeGraphEnabled;
+                    self.currentState.showsTimeFrameGraph = !self.currentState.showsTimeFrameGraph;
                     self.frameTimeGraphView.hidden = !self.frameTimeGraphEnabled;
                     break;
                 }
                 case MBXSettingsDebugToolsShowReuseQueueStats:
                 {
                     self.reuseQueueStatsEnabled = !self.reuseQueueStatsEnabled;
                     self.hudLabel.hidden = !self.reuseQueueStatsEnabled;
-                    self.mapInfoHUDEnabled = NO;
+                    self.zoomLevelOrnamentEnabled = NO;
```

Comment:
The use of `currentState` is not consistent. In some places you use the view controller properties (like here) and in other cases you use the properties in `currentState`